hnn_core/network_builder.py

```diff
@@ -631,27 +625,14 @@ def _record_spikes(self):
                 _PC.spike_record(gid, self._spike_times, self._spike_gids)
 
     # aggregate recording all the somatic voltages for pyr
-    def aggregate_currents(self):
-        """Aggregate somatic currents for Pyramidal cells."""
-        for cell in self.cells:
-            if cell.celltype in ('L5_pyramidal', 'L2_pyramidal'):
-                # iterate over dict_currents created in
-                # cell.record_current_soma()
-                for _, I_soma in cell.dict_currents.items():
-                    # self.current['L5_pyramidal_soma'] was created upon
-                    # in parallel, each node has its own NetworkBuilder()
-                    self.current['%s_soma' % cell.celltype].add(I_soma)
-
-    def aggregate_dipoles(self):
-        """Aggregate dipoles."""
+    def aggregate_data(self):
+        """Aggregate somatic currents, voltages, and dipoles."""
         for cell in self.cells:
             if cell.celltype in ('L5_pyramidal', 'L2_pyramidal'):
                 self.dipoles[cell.celltype].add(cell.dipole)
 
-    def aggregate_voltages(self):
-        """Organize voltage recordings into dictionary indexed by gid"""
-        for cell in self.cells:
             self._vsoma[cell.gid] = cell.rec_v
+            self._isoma[cell.gid] = cell.dict_currents
```

Comment:
dict_currents -> rec_i ?

Reply:
May as well. It is a dictionary of `h.Vector()` recordings unlike `vsoma` but anyone who is digging around this code can quickly figure that out.